src/components/Alert.tsx

```diff
@@ -77,11 +41,44 @@ const Alert: FC = () => {
     >
       {alert ? (
         <FadeTransition duration='slow' nodeRef={popupRef} onEntering={() => setDismiss(false)}>
-          {/* Specify a key to force the component to re-render and thus recalculate useSwipeToDismissProps when the alert changes. Otherwise the alert gets stuck off screen in the dismiss state. */}
-          <Popup {...alert} ref={popupRef} onClose={onClose} key={value}>
-            {value}
-            {buttons}
-          </Popup>
+          <PopupBase
+            anchorFromBottom
+            anchorOffset={36}
+            cssRaw={css.raw({
+              boxSizing: 'border-box',
+              background: 'panelBg',
+              border: '1px solid {colors.panelBorder}',
+              borderRadius: '8px',
+              zIndex: 'popup',
+              marginInline: 'auto',
+              left: 0,
+              right: 0,
+              width: 'max-content',
+            })}
+            ref={popupRef}
+            // Specify a key to force the component to re-render and thus recalculate useSwipeToDismissProps when the alert changes. Otherwise the alert gets stuck off screen in the dismiss state.
+            key={value}
+            circledCloseButton
+            showXOnHover
+            onClose={alert.showCloseLink && !isTouch ? onClose : undefined}
+            calculatedHeight={popupRef.current?.getBoundingClientRect().height || 50}
```

Comment:
`getBoundingClientRect` has performance implications because it forces a layout recalculation before returning up-to-date element dimensions and position. Therefore it should only be called on user action or in `useEffect` (or `useLayoutEffect` if critical for the initial paint).

Reply:
Moved into a `useEffect` (after moving the calculation into `PopupBase`): 869af85b3f63c61c957fbc15f7e72ab4b55623c7